DatadogInternal/Sources/Utils/Sampler.swift

```diff
@@ -43,5 +47,13 @@ internal struct DeterministicSampler: Sampling {
         self.shouldSample = shouldSample
     }
 
+    init(baseId: UInt64, samplingRate: Float) {
```

Comment:
question/ Is `baseId` a `TraceID`? If so, we could make it accept [`TraceID`](https://github.com/DataDog/dd-sdk-ios/blob/52842a99d3eced599fc59185f0c7c166aa00cdf1/DatadogInternal/Sources/NetworkInstrumentation/TraceID.swift#L9) to avoid ambiguity with accepting `UInt64` type. In fact, our `TraceID` is not as simple as `UInt64`. It has `idHi` and `idLo` making it unclear what should be the value passed here.